cpp/src/join/hash_join.cuh

```diff
@@ -322,8 +322,7 @@ get_base_hash_join_indices(
     CHECK_CUDA(stream);
 
     join_size = write_index.value();
-    estimated_size *= 2;
-  } while (estimated_size < join_size) ;
+  } while ((estimated_size < join_size) and (estimated_size *= 2)) ;
```

Comment:
This is potentially confusing. Mixing an assignment inside a logic check isn't very clear. Can this be done another way?

Reply:
I can create one more variable like current_estimated_size like below
```
current_estimated_size = estimated_size;
do {
...........
...........
current_estimated_size = estimated_size;
estimated_size *= 2;
} while (current_estimated_size < join_size) ;


```

Reply:
Or better way 
```
do {
if (join_size != 0){
estimated_size *= 2;
}
...........
...........
}while (estimated_size < join_size) ;
```

Reply:
moving with option 1 as it will not be dependent on join_size.